src/app/app-interactive-map/store/interactive-map.effects.ts

```diff
@@ -51,10 +51,12 @@ const preferredMapsByModel = {
 };
 
 const preferredSelector = <T>(
-    predicate: (item: T) => boolean,
-    preferredMap?: (item: T) => boolean,
-  ) => (items: T[]): T =>
-  items.find(predicate) || (preferredMap ? items.find(preferredMap) || items[0] : items[0]);
+  predicate: (item: T) => boolean,
+  preferredMap?: (item: T) => boolean,
+  modelHasMaps?: (item: T) => boolean,
+) => (items: T[]): T =>
+  items.find(predicate) || (items.filter(modelHasMaps).length > 0 ? items.filter(modelHasMaps)[0] : (items.find(preferredMap) ||
```

Comment:
`items.filter(modelHasMaps).length > 0 ? items.filter(modelHasMaps)[0]` is the same as `items.find(modelHasMaps)`.

Find returns the first element for which the filtering function returns true, if none fulfills the criteria, returns undefined.

By using `||` if the lefthand operation returns a falsy value, it calculates the right hand.